storage/storage.go

```diff
@@ -92,13 +96,30 @@ type Storer interface {
 
 	TaskRun(ctx context.Context, id int) (*TaskRunWithID, error)
 	// TaskRuns queries the storage for runs that match the filters.
+	// A limit value of 0 will return all results.
 	// The found results are passed in iterative manner to the callback
 	// function. When the callback function returns an error, the iteration
 	// stops.
 	// When no matching records exist, the method returns ErrNotExist.
 	TaskRuns(ctx context.Context,
 		filters []*Filter,
 		sorters []*Sorter,
+		limit int,
+		callback func(*TaskRunWithID) error,
+	) error
+
+	// TaskRunsWithInputURI queries the storage for runs that match the filters
+	// and contain an input with the given URI value.
+	// A limit value of 0 will return all results.
+	// The found results are passed in iterative manner to the callback
+	// function. When the callback function returns an error, the iteration
+	// stops.
+	// When no matching records exist, the method returns ErrNotExist.
+	TaskRunsWithInputURI(ctx context.Context,
```

Comment:
That the postgres storage  implementation requires a different query when a filter for the `FieldURI` field is specified, is specify to the postgres implementation.
This complexity and detail should be hidden in the storage interface.

The `TaskRuns()`  function could check if a filter for the `FieldURI` was passed, if yes it can use a different query.
What do you think, @StephenDiligent?

Reply:
I think this is a great idea, I'l make this suggested change